### ISSUE TITLE
Build issue with implicit fileno func def on linux

### ISSUE THREAD

**Issue by yuriy-yarosh** [CGNS/CGNS#810]
`ADF_internals.c` missing posix definition for strict fileno usage.

`externals/cgns/adf/ADF_internals.c` requires `#define _POSIX_SOURCE` before including `stdio.h` for fileno implicit function definition to actually be resolved, [over here](https://github.com/CGNS/CGNS/blob/develop/src/adf/ADF_internals.c#L169)

![Image](https://github.com/user-attachments/assets/218b1092-67e2-43d0-9f3a-63e1305e638e)

Related to [SU2/2409](https://github.com/su2code/SU2/issues/2409).

**Comment by yuriy-yarosh** [comment]
![pic](https://private-user-images.githubusercontent.com/6811933/399345480-f840d79a-b5b5-4b6e-8ed9-34949c09e7f7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU1ODIyODYsIm5iZiI6MTczNTU4MTk4NiwicGF0aCI6Ii82ODExOTMzLzM5OTM0NTQ4MC1mODQwZDc5YS1iNWI1LTRiNmUtOGVkOS0zNDk0OWMwOWU3ZjcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMzBUMTgwNjI2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGE4MTgyZGJlYmMxZDQ5ODRmZTc1ZWNjNWJkMjdiYjY5M2NmYTY4NjJmYTNmNzAzZDE5ZmIxNzE2YTVjMTgzYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.krLjf8U4wfcUJgi_ztMlOAQADVb0n7jFzD9Iel0FvYw)

**Comment by MicK7** [comment]
@yuriy-yarosh can you check if the current "develop" branch fix the issue ?

**Comment by yuriy-yarosh** [comment]
@MicK7 yup, it builds ![Image](https://github.com/user-attachments/assets/bfb9d552-f91a-43fc-998c-746b979413a4)

![Image](https://github.com/user-attachments/assets/84020a10-fa34-441c-b307-962571e53456)

![Image](https://github.com/user-attachments/assets/9eab2fe5-0f55-48df-85ac-080eead6b735)

**Comment by brtnfld** [comment]
Thanks, closing.
